sale_order_type/models/sale.py

```diff
@@ -25,12 +25,11 @@ def _compute_sale_type_id(self):
                     record.partner_id.with_context(
                         force_company=record.company_id.id
                     ).sale_type
-                    or self.partner_id.commercial_partner_id.with_context(
+                    or record.partner_id.commercial_partner_id.with_context(
                         force_company=record.company_id.id
                     ).sale_type
                 )
-                if sale_type:
-                    record.type_id = sale_type
+                record.type_id = sale_type
```

Comment:
Try this:
```suggestion
                record.type_id = sale_type or record.default_get(['type_id'])['type_id']
```
¿Same in account_move.py?